velox/CMakeLists.txt

```diff
@@ -11,31 +11,57 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-add_subdirectory(aggregates)
+
+# VELOX_BASE
 add_subdirectory(buffer)
 add_subdirectory(common)
-add_subdirectory(connectors)
 add_subdirectory(core)
-add_subdirectory(dwio)
-add_subdirectory(examples)
-add_subdirectory(exec)
-add_subdirectory(expression)
-add_subdirectory(external/date)
-add_subdirectory(external/md5)
-add_subdirectory(functions)
-add_subdirectory(parse)
 add_subdirectory(serializers)
 add_subdirectory(type)
 add_subdirectory(vector)
 add_subdirectory(row)
 add_subdirectory(flag_definitions)
+add_subdirectory(external/date)
+add_subdirectory(external/md5)
+#
+
+# examples depend on expression
+if((${VELOX_ENABLE_EXAMPLES} AND ${VELOX_ENABLE_EXPRESSION})
+   OR ${VELOX_BUILD_TESTING})
+  add_subdirectory(examples)
```

Comment:
IMO it will be confusing for a user to set `-DVELOX_ENABLE_EXAMPLES=ON` but then not get examples without any feedback.

Could we emit errors or warnings if the flags are inconsistent (potentially in a follow-up PR?)